File: frontend/src/services/userService.js
import apiClient from './apiClient';

const userService = {
  getMyProfile: () => apiClient.get('/users/me').then((res) => res.data),

  updateMyProfile: (payload) =>
    apiClient.put('/users/me', payload).then((res) => res.data),

  changePassword: (payload) =>
    apiClient.put('/users/me/password', payload).then((res) => res.data),
};

export default userService;

File: seedsanskriti_backend/main-service/src/main/java/com/seedsanskriti/service/UserService.java
package com.seedsanskriti.service;

import org.springframework.security.core.Authentication;

import com.seedsanskriti.dto.ChangePasswordRequest;
import com.seedsanskriti.dto.UpdateProfileRequest;
import com.seedsanskriti.dto.UserResponse;

public interface UserService {

    UserResponse getMyProfile(Authentication authentication);

    UserResponse updateMyProfile(UpdateProfileRequest request, Authentication authentication);

    String changePassword(ChangePasswordRequest request, Authentication authentication);
}
